### PR TITLE
postico 2.0.1

### PR DESCRIPTION
**Pull request by domdomegg**
_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
  - No, see https://github.com/orgs/Homebrew/discussions/4373
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
  - I think so, but not sure how to handle versions
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by razvanazamfirei** [comment]
I think we're in a situation where we have 2 stable versions; `postico 1.x` would be considered the [legacy cask](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask). Given our current limitations around renaming, I would suggest submitting the `postico 1.x` cask to [Homebrew/homebrew-cask-versions](https://github.com/Homebrew/homebrew-cask-versions), updating the current postico cask to version 2 and adding a caveat that the `postico 1.x` cask is available on `homebrew-cask-versions`.

File: Casks/postico@2.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version "2.0.1,9593"
+  sha256 "0a8d012ae67e51001c94c87f98274010524e94771d3a7009fb09eaa4e492e29a"
+
+  url "https://downloads.eggerapps.at/postico/postico-#{version.csv.second}.dmg"
+  name "Postico"
+  desc "GUI client for PostgreSQL databases"
+  homepage "https://eggerapps.at/postico2/"
+
+  livecheck do
+    url "https://releases.eggerapps.at/postico2/download"
```
Comment (miccal) [inline Casks/postico@2.rb:11]: This link will not work:
```
|-> curl -sLI https://releases.eggerapps.at/postico2/download
HTTP/1.1 405 Method Not Allowed
Server: TornadoServer/6.1
Content-Type: text/html; charset=UTF-8
Date: Fri, 31 Mar 2023 15:39:56 GMT
Content-Length: 87
```
When you try it, you get this:
```
audit for postico: failed
 - Version '2.0.1,9593' differs from '' retrieved by livecheck.
```

Comment (reitermarkus) [reply Casks/postico@2.rb:11]: @miccal, this should work once we re-implement the `--request GET` workaround again, see https://github.com/Homebrew/brew/pull/15099.

File: Casks/postico.rb [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+  sha256 "0a8d012ae67e51001c94c87f98274010524e94771d3a7009fb09eaa4e492e29a"
 
-  url "https://downloads.eggerapps.at/postico/postico-#{version.csv.second}.zip"
+  url "https://downloads.eggerapps.at/postico/postico-#{version.csv.second}.dmg"
   name "Postico"
   desc "GUI client for PostgreSQL databases"
-  homepage "https://eggerapps.at/postico/"
+  homepage "https://eggerapps.at/postico2/"
 
   livecheck do
```
Comment (razvanazamfirei) [inline Casks/postico.rb:10]: They have a [sparkle feed](https://releases.eggerapps.at/postico2/appcast.xml) but it doesn't seem up to date.

File: Casks/postico.rb [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
-  sha256 "8adcbe10a1a077739e56d5c2795b763d9e1c546b773c65fcfd9eb03e0edad8d7"
+  version "2.0.1,9593"
+  sha256 "0a8d012ae67e51001c94c87f98274010524e94771d3a7009fb09eaa4e492e29a"
 
-  url "https://downloads.eggerapps.at/postico/postico-#{version.csv.second}.zip"
+  url "https://downloads.eggerapps.at/postico/postico-#{version.csv.second}.dmg"
   name "Postico"
   desc "GUI client for PostgreSQL databases"
-  homepage "https://eggerapps.at/postico/"
+  homepage "https://eggerapps.at/postico2/"
```
Comment (reitermarkus) [inline Casks/postico.rb:8]:
```suggestion
  homepage "https://eggerapps.at/postico#{version.major}/"
```

**Review by reitermarkus — APPROVED** [review]

**Pull request merged**